core/src/messages/agent.rs

```diff
@@ -380,6 +380,21 @@ pub struct SpawnRequest {
     pub bearer_token: Option<String>,
 }
 
+#[serde_as]
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+pub struct ImageDownloadRequest {
+    pub cluster: ClusterName,
+
+    pub drone_id: DroneId,
+
+    pub image_url: String,
+}
+
+pub enum ResourceRequest {
+    Spawn(SpawnRequest),
```

Comment:
Since this is a request for a resource (noun), let's use nouns instead of verbs for the variants, e.g.

`ResourceRequest::Backend`, `ResourceRequest::Image`